training_model/one_file_train.py

```
@@ -289,148 +799,849 @@ def train(cfg: DictConfig) -> dict[str, int | Any]:
+        FileNotFoundError: If required paths are missing.
+        ConversionError: If GGUF conversion fails.
     """
     try:
-        llama_cpp_dir = os.path.abspath(cfg.paths.llama_cpp_dir)
-        conversion_script = os.path.join(llama_cpp_dir, "convert_hf_to_gguf.py")
+        llama_cpp_dir = Path(cfg.paths.llama_cpp_dir).resolve()
+        conversion_script = llama_cpp_dir / "convert_hf_to_gguf.py"
 
-        if not os.path.isdir(llama_cpp_dir):
+        if not llama_cpp_dir.is_dir():
             raise FileNotFoundError(f"llama.cpp directory not found: {llama_cpp_dir}")
```

Ruff (0.12.2)

Abstract `raise` to an inner function

(TRY301)